addons/binding/org.openhab.binding.xmltv/src/main/java/org/openhab/binding/xmltv/internal/handler/ChannelHandler.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.xmltv.internal.handler;
+
+import static org.openhab.binding.xmltv.internal.XmlTVBindingConstants.*;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.xmltv.internal.configuration.XmlChannelConfiguration;
+import org.openhab.binding.xmltv.internal.jaxb.Icon;
+import org.openhab.binding.xmltv.internal.jaxb.MediaChannel;
+import org.openhab.binding.xmltv.internal.jaxb.Programme;
+import org.openhab.binding.xmltv.internal.jaxb.Tv;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ChannelHandler} is responsible for handling information
+ * made available in regard of the channel and current program
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class ChannelHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(XmlTVHandler.class);
+
+    @NonNullByDefault({})
+    private ScheduledFuture<?> globalJob;
+
+    @Nullable
+    private MediaChannel mediaChannel;
+    @Nullable
+    private RawType mediaIcon = new RawType(new byte[0], RawType.DEFAULT_MIME_TYPE);
+    @Nullable
+    private RawType programIcon = new RawType(new byte[0], RawType.DEFAULT_MIME_TYPE);
+
+    public List<Programme> programmes = new ArrayList<>();
```

Comment:
Apparently the [XMLTV file format](http://wiki.xmltv.org/index.php/XMLTVFormat) uses this British "programme" variant of "program". Wouldn't it be better if the code sticks to using one variant? E.g. at line 86, 98 the US English "program" variant is used instead of "programme".

I'd prefer "program" myself too but if it's used in the format it's better to consequently use it that way. :disappointed: